modules/ROOT/pages/using-flows-for-service-orchestration.adoc

```diff
@@ -0,0 +1,128 @@
+* Connecting cloud and on-premise applications
+* Event processing where multiple services need to be composed
+
+
+== The Anatomy of a Flow
+
+A flow is in essence just a chain of Mule Components. Think of each component as a Lego block where a Flow is something you build with them. A flow also has a message source, the source of messages that are processed by the component chain.
+
+//image::flow.jpg[]
+
+== Flow Configuration
+
+A Flow is configured in XML using the <flow> element. Each flow has a name attribute, a message source (unless it's a private flow), one or more components and an optional error handler.
+
+*Basic Structure*
+
+[source,xml,linenums]
+----
+<flow name="">
+    - 0..1 Source
+    - 1..n Component(s)
+    - 0..1 Error Handler
+</flow>
+----
+
+Flows seem simple, yet can be quite powerful. In particular, when combined with expressions in Mule, they can allow for very sophisticated processing of the message contents. There are many elements that leverage expressions, including:
+
+* xref:about-operations.adoc[Operations]
+* xref:about-components.adoc[Core Components]
+* xref:scopes-concept.adoc[Scopes]
+
+== Example
+
+*Simple Book Order Processing Flow*
+
+[source,xml,linenums]
+----
+<flow name="orderHandling">
+  <file:listener directory="/myDirectory">
+    <scheduling-strategy>
+      <fixed-frequency/>
+    </scheduling-strategy>
+    <file:matcher filenamePattern="*.xml"/>
+  </file:listener>
+  <ee:transform>
+    <ee:message>
+      <ee:set-payload resource="createBookOrdersTransformation.dwl"/>
+    </ee:message>
+  </ee:transform>
+  <foreach collection="#[payload.orders]">
+    <choice>
+      <when expression="#[payload.order.'type' == 'book']">
+        <http:request method="POST" path="/book/inventory"/>
+        <http:request method="POST" path="/book/order"/>
+        <email:send toAddresses="#[payload.customer.email]">
+          <email:body >
+            <email:content ><![CDATA[#['Your order has been placed.']]]></email:content>
+          </email:body>
+        </email:send>
+        <db:insert>
+          <db:sql >INSERT INTO ORDERS(CUSTOMER, AMOUNT) VALUES (:id, :amount)</db:sql>
+          <db:input-parameters><![CDATA[#[{ id : payload.customer.id, amount: payload.order.total }]]]></db:input-parameters>
+        </db:insert>
+      </when>
+    </choice>
+  </foreach>
+  <error-handler >
+    <on-error-continue>
+      <jms:publish destination="failedOrders"/>
+    </on-error-continue>
+  </error-handler>
+</flow>
+----
+
+== Flow Behavior
+
+When a message is received or generated by the source the flow is started and the configured components are invoked in a chain in the same order as they are configured.  Some components accept child component elements, in this case these are processed before returning and continuing processing the main list.
+
+//image::flowrr.jpg[]
+
+== Private Flows
+
+A private flow is one that cannot be accessed from outside the JVM because it has no source defined.
```

Comment:
`outside the JVM ` -> `outside the Mule Application`

Reply:
Mule App